ui/src/app/components/dataflow-form/dataflow-form.component.html

```diff
@@ -1,38 +1,109 @@
+      <mat-form-field class="full-width" appearance="outline" matTooltip="Additional user labels to be specified for the job. Enter a json of &quot;key&quot;: &quot;value&quot; pairs. Example: {&quot;name&quot;: &quot;wrench&quot;, &quot;mass&quot;: &quot;1kg&quot;, &quot;count&quot;: &quot;3&quot; }."
+      [matTooltipPosition]="'right'">
+        <mat-label>Additional User Labels</mat-label>
+        <input matInput placeholder="Additional User Labels" type="text" formControlName="additionalUserLabels"/>
       </mat-form-field>
       <br>
-      <mat-form-field class="full-width" appearance="outline">
-        <mat-label>VPC Subnetwork</mat-label>
-        <input matInput placeholder="VPC Subnetwork Name" type="text" formControlName="subnetwork"/>
+      <mat-form-field class="full-width" appearance="outline" matTooltip="Name for the Cloud KMS key for the job. Key format is: projects/<project>/locations/<location>/keyRings /<keyring>/cryptoKeys/<key>. Omit this field to use Google Managed Encryption Keys."
+      [matTooltipPosition]="'right'">
+        <mat-label>KMS Key Name</mat-label>
+        <input matInput placeholder="KMS Key Name" type="text" formControlName="kmsKeyName"/>
       </mat-form-field>
+    </form>
+      <hr>
+    <form [formGroup]="presetFlagsForm" class="dataflow-form">
+      <h2>
+        Preset Flags 
+        <span><button mat-button class="configure" (click)="enablePresetFlags()" [disabled]="!disablePresetFlags">
+          <mat-icon>edit</mat-icon>
+          EDIT
+      </button></span>
+      </h2> 
+      <h5>These flags are set by SMT by default and <b>SHOULD NOT BE</b> modified unless running Dataflow in a non-standard configuration.</h5>
       <br>
-      <mat-form-field class="full-width" appearance="outline" matTooltip="Set maximum workers for the dataflow job(s). Default value: 50"
-      [matTooltipPosition]="'right'" matTooltipHideDelay="100000">
-        <mat-label>Max Workers</mat-label>
-        <input matInput placeholder="50" type="text" formControlName="maxWorkers"/>
+      <mat-form-field class="full-width" appearance="outline" matTooltip="Specify the project to run the dataflow job in."
+      [matTooltipPosition]="'right'">
+        <mat-label>Dataflow Project Id</mat-label>
+        <input matInput placeholder="Dataflow Project Id" type="text" formControlName="dataflowProjectId"/>
       </mat-form-field>
       <br>
-      <mat-form-field class="full-width" appearance="outline" matTooltip="Set initial number of workers for the dataflow job(s). Default value: 1"
-      [matTooltipPosition]="'right'" matTooltipHideDelay="100000">
-        <mat-label>Number of Workers</mat-label>
-        <input matInput placeholder="1" type="text" formControlName="numWorkers"/>
+      <mat-form-field class="full-width" appearance="outline" matTooltip="Specify the region to run the dataflow job in. It is recommended to keep the region same as Spanner region for performance."
```

Comment:
We should provide an example of what type of string needs to be input here.
Same goes for the template location as well.